infra/feast-operator/Makefile

```diff
@@ -162,7 +162,7 @@ docker-build: ## Build docker image with the manager.
 ## Build feast docker image.
 .PHONY: feast-ci-dev-docker-img
 feast-ci-dev-docker-img:
-	cd ./../.. && make build-feature-server-dev
+	cd ./../.. && make build-feature-server-dev-minimal
```

Comment:
@franciscojavierarceo Since we do not have larger runner to run kind + feast with all dependencies, either we need self-hosted runners or we need to use image with minimal dependencies installed in CI. 
Currently moving from [multicloud/Dockerfile.dev](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/feature_servers/multicloud/Dockerfile.dev) to [multicloud/Dockerfile](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/feature_servers/multicloud/Dockerfile) works in CI to run operator e2e test.

Reply:
My concern is that we miss potential breaks by using this minimal image. That's the risk here, right?

Cc @lokeshrangineni @tchughesiv @redhatHameed

Reply:
We will be using this image just for operator e2e test, feast functionality will be tested via integration and unit tests with all dependencies installed. 
@tchughesiv or someone more familiar with operator work can confirm if it's a big risk in operator testing or better way to handle this.

Reply:
this change needs to be reverted

Reply:
@ntkathole this must remain `make build-feature-server-dev`